index.js

```diff
@@ -14,31 +50,21 @@ var promiseResolver = require('promise-resolver');
  * @return  {Function(arguments)} Arguments to pass to `func`. This function will in turn
  *                                return a Promise (Node >= 0.12) or call the callbacks.
  */
+module.exports = createRunAsync(false);
 
-module.exports = function (func, cb) {
-  return function () {
-    var async = false;
-    var deferred = promiseResolver.defer(cb);
-
-    try {
-      var answer = func.apply({
-        async: function () {
-          async = true;
-          return deferred.cb;
-        }
-      }, Array.prototype.slice.call(arguments));
-
-      if (!async) {
-        if (isPromise(answer)) {
-          answer.then(deferred.resolve, deferred.reject);
-        } else {
-          deferred.cb(null, answer);
-        }
-      }
-    } catch (e) {
-      deferred.cb(e);
-    }
-
-    return deferred.promise;
-  }
-};
+/**
+ * Return a function that will run a function asynchronously or synchronously.
+ * Supports functions with callback as their las input parameter
+ *
+ * example:
+ * function wrappedFunction(a, b, cb) {
+ *   cb(null, a + b);
+ * }
+ * runAsync.cb(wrappedFunction, callback)(...args);
+ *
+ * @param   {Function} func  Function to run
+ * @param   {Function} cb    Callback function passed the `func` returned value
+ * @return  {Function(arguments)} Arguments to pass to `func`. This function will in turn
+ *                                return a Promise (Node >= 0.12) or call the callbacks.
+ */
+module.exports.cb = createRunAsync(true);
```

Comment:
That's also very complex.

Just make that function check the input function length and inject the callback function as parameter. This function could just wrap the base `runAsync`.